src/binary_to_text/binary_encoder.ml

```diff
@@ -0,0 +1,713 @@
+(* SPDX-License-Identifier: AGPL-3.0-or-later *)
+(* Copyright © 2021-2024 OCamlPro *)
+(* Written by the Owi programmers *)
+
+open Binary
+open Syntax
+open Types
+
+let end_opcode = '\x0B'
+
+let magic = "\x00\x61\x73\x6d"
+
+let version = "\x01\x00\x00\x00"
+
+let add_char c buf = Buffer.add_char buf c
```

Comment:
Why did you change the order ? I believe it would allow to make the code more clearer to simply remove this function and use `Buffer.add_char buf c` in many places.

Reply:
`buf` is now the 1st argument of all functions.
See the following reviews ...